editor/editor_node.cpp

```diff
@@ -1215,6 +1215,7 @@ void EditorNode::_dialog_action(String p_file) {
 						accept->popup_centered_minsize();
 						return;
 					}
+					ml = Ref<TileSet>(memnew(TileSet));
```

Comment:
This whole logic is a bit convoluted, why try to load the existing file to only *afterwards* check that a merge was requested, and if not clear the loaded stuff or initialize it?

It should just be:
```
Ref<TileSet> tileset;   // not ml, it's not a MeshLibrary
if (FileAccess::exists(p_file) && file_export_lib_merge->is_pressed()) {
    tileset = ResourceLoader::load(p_file, "TileSet");
    if (tileset.is_null()) {
        current_option = -1;
        accept->get_ok()->set_text(TTR("I see.."));
        accept->set_text(TTR("Can't load TileSet for merging!"));
        accept->popup_centered_minsize();
        return;
    }
} else { // create or overwrite
    tileset = Ref<TileSet>(memnew(TileSet));
}
```

Reply:
I assume it was this way because overwriting the existing tileset might not work, or it's somehow better to load and clear (the existing tiles in the tileset). But if overwriting works, I agree it should be like that.

Reply:
I think load and clear is the same as write anew. I guess the code was like that because there was no merging option initially, it was the only behaviour.

Reply:
Bump :)

Reply:
```
if (what == GitHub::NOTIFICATION_BUMP)
    commit_and_push();
```